suse_ha-formula/suse_ha/macros.jinja

```diff
@@ -17,6 +17,37 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
 -#}
 
 {%- from './map.jinja' import resources_dir, management -%}
+
+{%- macro ha_resource_update(resource, origin) -%}
+ha_resource_update_{{ resource }}:
+  cmd.run:
+    - name: crm configure load xml update {{ resources_dir }}/{{ resource }}.xml
+    - onchanges:
+        - ha_{{ origin }}_file_{{ resource }}
+{#- to-do: somehow require this but allow for individual run of suse_ha.resources ?
+    - require:
+        - pacemaker.service
+#}
+{%- endmacro -%}
+
+{%- macro ha_constraint(cid, ctype, cscore=None, crscs=[], csets={}) -%}
+ha_constraint_file_{{ cid }}:
+  file.managed:
+    - name: {{ resources_dir }}/{{ cid }}.xml
+    - source: salt://suse_ha/files/cib/constraint.xml.j2
+    - template: jinja
+    - context:
+        cid: {{ cid }}
+        ctype: {{ ctype }}
+        cscore: {{ cscore if cscore is not none else 'null' }}
```

Comment:
I'm not sure how the _string_ `null` behaves in `is not none` checks. Did you test (or know from previous usage) that it will work?

Reply:
Hi, `null` in YAML translates to `None` in Python. If I put `None` in YAML, it would be a string with the content "None" in Python/Jinja.

Reply:
Thanks, that explains it :-)